internal/controller/storageclient_controller.go

```diff
@@ -210,6 +218,23 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 			if err := r.reconcileClusterResourceQuota(clusterResourceQuotaSpec); err != nil {
 				return reconcile.Result{}, err
 			}
+		case "CephConnection":
+			if utils.DelegateCSI {
+				cephConnection := &csiopv1a1.CephConnection{}
+				cephConnection.Name = r.storageClient.Name
+				cephConnection.Namespace = r.OperatorNamespace
+				if err := r.createOrUpdate(cephConnection, func() error {
+					if err := r.own(cephConnection); err != nil {
+						return fmt.Errorf("failed to own cephConnection resource: %v", err)
+					}
+					if err := json.Unmarshal(eResource.Data, &cephConnection.Spec); err != nil {
+						return fmt.Errorf("failed to unmarshall cephConnectionSpec: %v", err)
+					}
```

Comment:
This PR already established a pattern where the unmarshaling happens outside and before createOrUpdate, can you please align here as well?

Reply:
We are unmarshaling two shapes, one is of map[string]string and another is the spec specific to the resource that we are interested and I couldn't club both.